grammars/git commit message.cson

```diff
@@ -9,33 +9,35 @@
 'patterns': [
   {
     'begin': '\\A(?!#)'
-    'end': '(?=# Please enter the commit message)'
+    'end': '^(?<!\\A)(?=# Please enter the commit message)'
     'name': 'meta.scope.message.git-commit'
     'patterns': [
       {
-        'begin': '\\G'
-        'end': '^(?!\\G)'
-        'patterns': [
-          {
-            'captures':
-              '1':
-                'name': 'keyword.other.$2.git-commit'
-            'match': '\\G((fixup|squash)!)\\s*'
-          }
-          {
-            'match': '.{66,}$'
-            'name': 'invalid.illegal.line-too-long.git-commit'
-          }
-          {
-            'match': '.{51,}$'
-            'name': 'invalid.deprecated.line-too-long.git-commit'
-          }
-        ]
+        'captures':
+          '1':
+            'name': 'keyword.other.$2.git-commit'
+        'match': '\\G((fixup|squash)!)\\s*'
+      }
+      {
+        'match': '^#.*$'
+        'name': 'comment.line.number-sign.git-commit'
+      }
+      {
+        'match': '\\A(?!#).{70,}'
+        'name': 'invalid.illegal.line-too-long.git-commit'
+      }
+      {
+        'match': '\\A(?!#).{51,}'
+        'name': 'invalid.deprecated.line-too-long.git-commit'
```

Comment:
Many themes will highlight the `deprecated` scope to warn users, with more subtle colouring than what the `invalid` scope uses (since it's less an error than a warning). Summary lines over 50 characters are still legal, but discouraged.